Extensions/Leaderboards/leaderboardstools.ts

```diff
@@ -120,61 +146,161 @@ namespace gdjs {
+            _leaderboardViewIframeLoading = false;
+
+            break;
         }
       };
 
-      export const displayLeaderboard = async function (
+      const onError = function (
         runtimeScene: gdjs.RuntimeScene,
-        leaderboardId: string
+        message: string
       ) {
-        const gameId = gdjs.projectData.properties.projectUuid;
-        const targetUrl = `https://liluo.io/games/${gameId}/leaderboard/${leaderboardId}`;
-        if (!(await checkLeaderboardAvailability(targetUrl))) {
-          logger.error('Leaderboard data could not be fetched, doing nothing');
-          return;
-        }
+        logger.error(message);
+        _leaderboardViewIframeErrored = true;
+        _leaderboardViewIframeLoading = false;
+        closeLeaderboardView(runtimeScene);
+      };
 
-        if (_leaderboardViewIframe) {
-          _leaderboardViewIframe.src = targetUrl;
-        } else {
-          const domElementContainer = runtimeScene
-            .getGame()
-            .getRenderer()
-            .getDomElementContainer();
-          if (!domElementContainer) {
-            logger.error(
-              "The div element covering the game couldn't be found, the leaderboard cannot be displayed."
+      const resetErrorTimeout = (runtimeScene: gdjs.RuntimeScene) => {
+        if (_errorTimeoutId) clearTimeout(_errorTimeoutId);
+        _errorTimeoutId = setTimeout(() => {
+          if (!_leaderboardViewIframeLoaded) {
+            onError(
+              runtimeScene,
+              'Leaderboard page did not send message in time. Closing leaderboard view.'
             );
-            return;
           }
-          const iframe = document.createElement('iframe');
-
-          iframe.src = targetUrl;
-          iframe.id = 'leaderboard-view';
-          iframe.style.position = 'absolute';
-          iframe.style.pointerEvents = 'all';
-          iframe.style.top = '0px';
-          iframe.style.height = '100%';
-          iframe.style.left = '0px';
-          iframe.style.width = '100%';
-          iframe.style.border = 'none';
-          _leaderboardViewIframe = iframe;
-          if (typeof window !== 'undefined') {
-            _leaderboardViewClosingCallback = (event: MessageEvent) => {
-              receiveMessage(runtimeScene, event);
-            };
-            (window as any).addEventListener(
-              'message',
-              _leaderboardViewClosingCallback,
-              true
+        }, 5000);
+      };
+
+      export const displayLeaderboard = function (
+        runtimeScene: gdjs.RuntimeScene,
+        leaderboardId: string,
+        displayLoader: boolean
+      ) {
+        _leaderboardViewIframeErrored = false;
+        _leaderboardViewIframeLoaded = false;
+        _leaderboardViewIframeLoading = true;
+        const gameId = gdjs.projectData.properties.projectUuid;
+        const targetUrl = `http://localhost:3001/games/${gameId}/leaderboard/${leaderboardId}?inGameEmbedded=true&dev=true`;
```

Comment:
Add a TODO when you leave development urls ;) Easier to spot before merging

Reply:
😓